### PR TITLE
Add lineage link

### PR DESCRIPTION
**Pull request by danwom**
- Added a lineage section in the table details side bar. This link can be enabled and customized by overwriting the default config values.
- Cleaned up some global styles by nesting them inside their component classes.
- Removed deprecated 'nav_link' file.
- Added comments for static configs.

### CODE REVIEW

File: amundsen_application/static/js/components/common/Feedback/styles.scss [hunk 1]
```diff
@@ -5,67 +5,58 @@
...
+  }
+
+  .feedback-header {
+    margin-bottom: 8px;
+  }
+
+  .btn-group {
+    margin: 8px auto 16px;
+  }
 }
```
Comment (danwom) [inline amundsen_application/static/js/components/common/Feedback/styles.scss]: These css rules were being applied to all images so I nested the rules and deleted a few that were unused.

**Comment by danwom** [comment]
👀 @ttannis @feng-tao

**Comment by feng-tao** [comment]
Given we are open source now, I would prefer our PR description is more descriptive as not everyone has the same context on the pr.

And I wonder how external people could use the feature. Does it require to have a lineage service which visualizes the data somewhere else? cc @jinhyukchang . I wonder whether Apache Atlas supported this feature.

And have we decided not using the column linage graph?

**Comment by feng-tao** [comment]
It seems that you are removing some existing nav_links which I think we should mention in the pr description as well.

File: amundsen_application/static/config/config-default.ts [hunk 2]
```diff
@@ -15,6 +15,13 @@ const configDefault: AppConfig = {
     key: 'default-key',
     sampleRate: 100,
   },
+  lineage: {
```
Comment (feng-tao) [inline amundsen_application/static/config/config-default.ts]: it would be good if we could add a comment for what the config is used for. In fact, it would be good to have a comment for all the config here(could be a different pr)

Comment (danwom) [reply amundsen_application/static/config/config-default.ts]: I'll add some comments to the type definitions in config.types.ts.
